lib/modules/datasource/dotnet-sdk/index.ts

```diff
@@ -0,0 +1,83 @@
+import is from '@sindresorhus/is';
+import { cache } from '../../../util/cache/package/decorator';
+import type { HttpResponse } from '../../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { DotnetSdkReleases, DotnetSdkReleasesIndex } from './types';
+
+export class DotnetSdkDatasource extends Datasource {
+  static readonly id = 'dotnet-sdk';
```

Comment:
i think this should support runtime too. so we can use this datasource for dotnet runtime updates later.

the adoption datasource for java is a good sample for jre and jdk

Reply:
I'll rename this to `dotnet` instead of `dotnet-sdk`?

Reply:
Renamed to `dotnet` and added support for `dotnet-sdk` and `dotnet-runtime` dependencies.